src/Abstractions/NexusMods.Abstractions.Jobs/AJobWorker.cs

```diff
@@ -0,0 +1,69 @@
+using JetBrains.Annotations;
+
+namespace NexusMods.Abstractions.Jobs;
+
+[PublicAPI]
+public abstract class AJobWorker : IJobWorker
+{
+    IJob IJobWorker.Job => Job;
+    internal AJob Job { get; }
+
+    protected AJobWorker(AJob job)
+    {
+        Job = job;
+    }
+
+    protected abstract Task<JobResult> ExecuteAsync(CancellationToken cancellationToken);
+
+    public Task StartAsync(CancellationToken cancellationToken = default)
+    {
+        
+
+        throw new NotImplementedException();
+    }
+
+    public Task PauseAsync(CancellationToken cancellationToken = default)
+    {
+        throw new NotImplementedException();
+    }
+
+    public Task CancelAsync(CancellationToken cancellationToken = default)
+    {
+        throw new NotImplementedException();
+    }
+
+    protected JobResult FailJob(Exception exception)
+    {
+        throw new NotImplementedException();
+    }
+
+    protected JobResult FailJob(string message)
+    {
+        throw new NotImplementedException();
+    }
+
+    protected JobResult CompleteJob<TData>(TData data)
+    {
+        throw new NotImplementedException();
+    }
```

Comment:
Shouldn't these `NotImplementedException()` method just be marked as abstract instead?

Reply:
Not really, I just haven't implemented them yet.